primitives/consensus/common/src/import_queue/basic_queue.rs

```diff
@@ -40,9 +41,29 @@ pub struct BasicQueue<B: BlockT, Transaction> {
 	manual_poll: Option<Pin<Box<dyn Future<Output = ()> + Send>>>,
 	/// A thread pool where the background worker is being run.
 	pool: Option<futures::executor::ThreadPool>,
+	pool_guard: Arc<(Mutex<usize>, Condvar)>,
 	_phantom: PhantomData<Transaction>,
 }
 
+impl<B: BlockT, Transaction> Drop for BasicQueue<B, Transaction> {
+	fn drop(&mut self) {
+		drop(self.pool.take());
+		// Flush the queue and close the receiver to terminate the future.
+		let _ = self.sender.unbounded_send(ToWorkerMsg::Shutdown);
```

Comment:
I think we should just close it: https://docs.rs/futures/0.3.4/futures/channel/mpsc/struct.UnboundedSender.html#method.close_channel

Reply:
```suggestion
		self.sender.close_channel();
```